### PR TITLE
Eliminate static options access for central ee

### PR DESCRIPTION
**Pull request by ajreynol**
Also refactors TheoryEngine to not maintain its own reference to logic info and an accessor to it.

Towards eliminating option scopes.

### CODE REVIEW

File: src/theory/theory_engine.cpp [hunk 1]
```diff
@@ -469,7 +472,8 @@ void TheoryEngine::check(Theory::Effort effort) {
       for (TheoryId theoryId = THEORY_FIRST; theoryId < THEORY_LAST; ++theoryId) {
         if( theoryId!=THEORY_QUANTIFIERS ){
           Theory* theory = d_theoryTable[theoryId];
-          if (theory && d_logicInfo.isTheoryEnabled(theoryId)) {
+          if (theory && logicInfo().isTheoryEnabled(theoryId))
```
Comment (nafur) [inline src/theory/theory_engine.cpp:475]:
```suggestion
          if (theory && isTheoryEnabled(theoryId))
```

File: src/theory/theory_engine.cpp [hunk 2]
```diff
@@ -885,9 +910,7 @@ void TheoryEngine::assertToTheory(TNode assertion, TNode originalAssertion, theo
 
   // determine the actual theory that will process/explain the fact, which is
   // THEORY_BUILTIN if the theory uses the central equality engine
-  TheoryId toTheoryIdProp = (Theory::expUsingCentralEqualityEngine(toTheoryId))
-                                ? THEORY_BUILTIN
-                                : toTheoryId;
+  TheoryId toTheoryIdProp = theoryExpPropagation(toTheoryId);
```
Comment (nafur) [inline src/theory/theory_engine.cpp:913]: It seems this changes behaviour?

File: src/theory/theory_engine.cpp [hunk 3]
```diff
@@ -1477,10 +1507,8 @@ TrustNode TheoryEngine::getExplanation(
   Node conclusion = explanationVector[0].d_node;
   // if the theory explains using the central equality engine, we always start
   // with THEORY_BUILTIN.
-  if (Theory::expUsingCentralEqualityEngine(explanationVector[0].d_theory))
-  {
-    explanationVector[0].d_theory = THEORY_BUILTIN;
-  }
+  explanationVector[0].d_theory =
```
Comment (nafur) [inline src/theory/theory_engine.cpp:1510]: Same here.

File: src/theory/theory_engine.cpp (hunk 2 above)
Comment (ajreynol) [reply src/theory/theory_engine.cpp:913]: It should be the same I believe?

Comment (nafur) [reply src/theory/theory_engine.cpp:913]: It now considers the `eeMode` option and the result of `usesCentralEqualityEngine()`... if this is intentional, everything is fine.

Comment (ajreynol) [reply src/theory/theory_engine.cpp:913]: `expUsingCentralEqualityEngine` used tocall to `usesCentralEqualityEngine`, so this was factored out and moved to EqualiyEngineManagerCentral.  I believe the code should be equivalent still.

Comment (nafur) [reply src/theory/theory_engine.cpp:913]: Ah right. Sorry :)

**Review by nafur — APPROVED** [review]
LGTM.

**Pull request merged**